src/events/dns.js

```diff
@@ -0,0 +1,239 @@
+const eventInterface = require('../event.js');
+const errorCode = require('../proto/error_code_pb.js');
+
+// BLACKLIST
+const URL_BLACKLIST = {
+    'tc.epsagon.com': 'endsWith',
+};
+/**
+ * Checks if a URL is in the blacklist
+ * @param {string} url The URL to check
+ * @param {string} path The Path to check (optional)
+ * @returns {boolean} True if it is in the blacklist, False otherwise
+ */
+function isBlacklistURL(url, path) {
+    return Object.keys(URL_BLACKLIST).some((key) => {
+        if (typeof URL_BLACKLIST[key] === typeof (() => {})) {
+            return URL_BLACKLIST[key](url, key, path);
+        }
+        return url[URL_BLACKLIST[key]](key);
+    });
+}
+
+const rrtypesMethods = {
+    resolveAny: 'ANY',
+    resolve6: 'AAAA',
+    resolve4: 'A',
+    resolveCname: 'CNAME',
+    resolveMx: 'MX',
+    resolveNaptr: 'NAPTR',
+    resolveNs: 'NS',
+    resolvePtr: 'PTR',
+    resolveSoa: 'SOA',
+    resolveSrv: 'SRV',
+    resolveTxt: 'TXT',
+};
+
+const initialDnsEvent = (name) => {
+    const startTime = Date.now();
+    const resource = new serverlessEvent.Resource([
+        'dns',
+        'dns',
+        name,
+    ]);
+    const dnsEvent = new serverlessEvent.Event([
+        `dns-${uuid4()}`,
+        utils.createTimestampFromTime(startTime),
+        null,
+        'dns',
+        0,
+        errorCode.ErrorCode.OK,
+    ]);
+    dnsEvent.setResource(resource);
+    return { dnsEvent, startTime };
+};
+
+const endDurationAndAddCallbackMetadata = (dnsEvent, startTime, err, metadata) => {
+    dnsEvent.setDuration(utils.createDurationTimestamp(startTime));
+    if (err) {
+        eventInterface.setException(dnsEvent, err);
+    } else {
+        eventInterface.addToMetadata(dnsEvent, { ...metadata });
+    }
+};
+
+const getRrtypeArguments = (arg1, arg2, arg3, functionName) => {
+    const hostname = arg1;
+    let rrtype = arg2;
+    let callback = arg3;
+    if (!arg3) {
+        if (functionName) {
+            rrtype = Object.values(rrtypesMethods).find(type => functionName.toLocaleLowerCase().includes((`query${type}`.toLocaleLowerCase())));
+        }
+        if (!rrtype) rrtype = rrtypesMethods.resolve4;
+        callback = arg2;
+    }
+    return { hostname, rrtype, callback };
+};
+
+const getArguments = (arg1, arg2, arg3) => {
+    const hostname = arg1;
+    let options = arg2;
+    let callback = arg3;
+    if (!arg3) {
+        options = undefined;
+        callback = arg2;
+    }
+    return { hostname, options, callback };
+};
+
+const buildParams = (arg1, arg2, arg3) => {
+    if (!arg3) {
+        return [arg1, arg2];
+    }
+    return [arg1, arg2, arg3];
+};
+
+const wrapDnsResolveFunction = original => (arg1, arg2, arg3) => {
+    let patchedCallback;
+    let clientRequest;
+    let options;
+    const { hostname, rrtype, callback } = getRrtypeArguments(arg1, arg2, arg3, original.name);
+    if (!callback) return original.apply(this, buildParams(arg1, arg2, arg3));
```

Comment:
use {}

Reply:
wait also why? we can still have our callback set up and wait for the response no?